packages/react/src/components/ComboBox/ComboBox.stories.js

```diff
@@ -79,6 +79,33 @@ export const Default = () => (
   </div>
 );
 
+export const AllowCustomValue = () => {
+  const filterItems = (menu) => {
+    return menu?.item?.toLowerCase().includes(menu?.inputValue?.toLowerCase());
+  };
+  return (
+    <div style={{ width: 300 }}>
+      <ComboBox
+        allowCustomValue
+        shouldFilterItem={filterItems}
+        onChange={(e) => {
+          console.log(e);
+        }}
+        id="carbon-combobox"
+        items={['Apple', 'Orange', 'Banana', 'Pineapple', 'Raspberry', 'Lime']}
+        downshiftProps={{
+          onStateChange: () => {
+            console.log('the state has changed');
+          },
+        }}
+        // itemToString={(item) => (item ? item.text : '')}
```

Comment:
do we want to keep this commented out here? @tw15egan 

Other than this - looks good!

Reply:
Removed 👍🏻